### DISCUSSION TITLE
Check arrays field size with gt and lt

### DISCUSSION THREAD

**Discussion by Rutenioh** [stalniy/casl#953] [General]
Hello,

I am trying to add a rule using MongoDB conditions to check if an array is greater or less than a specific number.

I have done several tests, the first one being that the field is greater than the number:

```
"conditions" : {
    "arrayField" : {
        "$lte" : 5
    }
}
```
This won't work because it checks the value, not the size.

I then tried using $expr and $where, but I can't find a way to implement it.
Our way to do the is:
```
ForbiddenError.from(ability).throwUnlessCan(action,
      subjectCasl(subject, subjectDTO || {}),
      field));
```
 We load the rule definition from db and use it this way. The rule definition is:
```
DefinitionItem {
    action: string;
    conditions?: Record<string, unknown>;
    fields?: string[];
    inverted?: boolean;
    reason?: string;
    subject: string;
}
```
and we create the ability:
```
const rule = //load from db
const ability = createMongoAbility(rule.definition)
```

Thank you, and if you need more information, please let me know.

**Comment by stalniy** [comment]
you have 2 options:
1. implement own custom operator -> https://casl.js.org/v6/en/advanced/customize-ability#extend-conditions-with-custom-operators but then it will work only in runtime
2. add support for `$size` MongoDB operator -> https://www.mongodb.com/docs/manual/reference/operator/query/size/ 

The 2nd option doesn't support ranges but you can do this:

```ts
can('read', 'Type', { arrayField: { $size: 2 } });
can('read', 'Type', { arrayField: { $size: 3 } });
can('read', 'Type', { arrayField: { $size: 4 } });
```

This means that user can read Type if its arrayField's length between 2 and 4 inclusively
